system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/aws/test/e2e/S3ProvisionEndToEndTest.java

```diff
@@ -0,0 +1,354 @@
+import static org.eclipse.edc.aws.test.e2e.EndToEndTestCommon.createAsset;
+import static org.eclipse.edc.aws.test.e2e.EndToEndTestCommon.createConsumerSecret;
+import static org.eclipse.edc.aws.test.e2e.EndToEndTestCommon.createContractDefinition;
+import static org.eclipse.edc.aws.test.e2e.EndToEndTestCommon.createPolicy;
+import static org.eclipse.edc.aws.test.e2e.EndToEndTestCommon.getAgreementId;
+import static org.eclipse.edc.aws.test.e2e.EndToEndTestCommon.initiateNegotiation;
+import static org.eclipse.edc.aws.test.e2e.EndToEndTestCommon.initiateTransfer;
+import static org.eclipse.edc.aws.test.e2e.EndToEndTestCommon.waitForNegotiationState;
+import static org.eclipse.edc.aws.test.e2e.EndToEndTestCommon.waitForProviderTransferState;
+import static org.eclipse.edc.aws.test.e2e.EndToEndTestCommon.waitForTransferState;
+import static org.eclipse.edc.connector.controlplane.contract.spi.types.negotiation.ContractNegotiationStates.FINALIZED;
+import static org.eclipse.edc.connector.controlplane.transfer.spi.types.TransferProcessStates.COMPLETED;
+import static org.eclipse.edc.connector.controlplane.transfer.spi.types.TransferProcessStates.DEPROVISIONED;
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.IAM;
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.S3;
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.STS;
+
+@EndToEndTest
+@Testcontainers
+class S3ProvisionEndToEndTest {
+
+    private static final DockerImageName LOCALSTACK_DOCKER_IMAGE = DockerImageName.parse("localstack/localstack:4.2.0");
+
+    private static final String SYSTEM_PROPERTY_AWS_ACCESS_KEY_ID = "aws.accessKeyId";
+    private static final String SYSTEM_PROPERTY_AWS_SECRET_ACCESS_KEY = "aws.secretAccessKey";
+
+    private final String fileContent = "Hello, world!";
+    private final String region = "eu-central-1";
+
+    private final String sourceBucket = "source-bucket";
+    private final String sourceObjectName = "source.txt";
+    private final String sourceUser = "source-user";
+    private final String sourceUserPolicyName = "source-user-policy";
+    private String sourceAccessKeyId;
+    private String sourceSecretAccessKey;
+
+    private final String destinationBucket = "destination-bucket";
+    private final String destinationObjectName = "transferred.txt";
+    private final String destinationUser = "destination-user";
+    private final String destinationUserPolicyName = "destination-user-policy";
+    private String destinationAccessKeyId;
+    private String destinationSecretAccessKey;
+
+    @Container
+    private static final LocalStackContainer LOCALSTACK_CONTAINER =
+            new LocalStackContainer(LOCALSTACK_DOCKER_IMAGE).withServices(S3, IAM, STS);
+
+    @RegisterExtension
+    private static final RuntimeExtension PROVIDER = new RuntimePerClassExtension(
+            new EmbeddedRuntime("provider", getAdditionalModules())
```

Comment:
these runtimes shouldn't have the "*-copy" modules, otherwise those are the provisioner and pipeline modules being used, please remove them from the runtime build file and add them to the `S3CopyEndToEndTest` runtimes